Сeilometer integration

r7vme opened the pull request:
- HA and simple schemas
- Tested on CentOS 6.4, also it works on Ubuntu with some package dependencies fixes.
- SQL db backend. We've succeded with stress tests on ceilometer with mysql.
- Ceilometer is optional and by default disabled.

Commit:
Initial ceilometer integration

  * simple (controller+compute) deployment
  * tested on CentOS 6.4 support
  * tested w/ Havana and Grizzly
  * qpid support
  * NO High-Availability
  * NO logging
  * Needed some configuration of cinder, ?glance, etc.

Commit:
Change ini-group in ceilometer.conf for api and database parameters

r7vme committed:
Merge pull request #3 from vsaharov/master-ceilometer

Change ini-group in ceilometer.conf for api and database parameters

Commit:
Fixes PDT-1. Adds iptables rule for ceilometer.

dnikishov committed:
Ceilometer params fixes

Change-Id: If7d5f74c7a2b4499eb8579adfeee83f5de3c833e

Commit:
Merge branch 'master-ceilometer' into havana-ceilometer

Conflicts:
	deployment/puppet/openstack/manifests/compute.pp
	deployment/puppet/openstack/manifests/db/mysql.pp
	deployment/puppet/osnailyfacter/manifests/cluster_simple.pp

r7vme committed:
Merge pull request #5 from dnikishov/havana-ceilometer

Havana ceilometer

Commit:
Adds ceilometer logging.

 - ceilometer uses facility SYSLOG
 - also deleted some deprecated flags from config

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/456/

dnikishov committed:
fix ceilometer_metering_secret on compute

Change-Id: I796197ae005439dbe0ed52c2bec50ec2f20197b4

r7vme committed:
Merge pull request #6 from dnikishov/havana-ceilometer

fix ceilometer_metering_secret on compute

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/459/

dmitryilyin commented on deployment/puppet/ceilometer/manifests/agent/compute.pp:
```diff
@@ -0,0 +1,113 @@
+    ],
+  }
+
+  file_line {
```
Was there a reason to use File_line instead of Nova_config for 'nova-notification-driver-common' and 'nova-notification-driver-ceilometer' ?

r7vme replied:
it tries to add two equal options "notification_driver" this was cause, but we could refactor this piece in favor notification_driver=driver1,driver2, it also supported and use nova_config here.

dmitryilyin replied:
Well... it's ok if you know what and why you are doing)

dmitryilyin commented:
Looks very much like some other openstack manifests. And most likely will suffer from the same Puppet containment problem I've described here https://docs.google.com/a/mirantis.com/document/d/1VopEsyBq3SqErX5FI5FwWl02VYqmL6lS4O_yCQ6Air4/edit

For example, this may look like it will force database creation striclty before  other ceilometer classes
Class['openstack::db::mysql'] -> Class['openstack::ceilometer']
But it may not be the case. Dependencies between classes doesn't lead to dependencies between resources inside subclasses. Ad mysql database may or may not try to be created after the start of ceilomemter services.

But there are also a lot of other interdependencies betwen components so this may be already fixed... Connections are too complex for me to untangle without running and making --graph.

Al least expect to meet problems with mysql database/user/permission sometimes not being created when services are started and try to double check this.

Commit:
Refactored piece for notification_driver

 * now use nova_config instead file_line
 * also fix problem, when db sync fails because of no logging.conf

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/462/

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_pullrequest_systest/33/

vsaharov committed:
Added manifests for alarm service of ceilometer on CentOS (tested) and Ubuntu (not tested).

r7vme committed:
Merge pull request #8 from vsaharov/havana-ceilometer

Added manifests for alarm service of ceilometer on CentOS (tested) and U...

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/515/

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_pullrequest_systest/77/

monroid commented:
retest this please

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/516/

monroid commented:
retest this please

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/517/

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_pullrequest_systest/78/

mirantis-jenkins commented:
Merged build started.

monroid commented:
retest this please

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/520/

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_pullrequest_systest/82/

monroid commented:
retest this please

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/522/

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_pullrequest_systest/84/

mirantis-jenkins commented:
Merged build triggered.

monroid commented:
retest this please

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build triggered.

mirantis-jenkins commented:
Merged build started.

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/542/

mirantis-jenkins commented:
Merged build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_pullrequest_systest/104/

Commit:
HA mode support.

  - c-agent-central under pacemaker in active/passive mode.
  - other services in multiple instances.

mirantis-jenkins commented:
Build triggered.

mirantis-jenkins commented:
Build started.

mirantis-jenkins commented:
Build triggered.

mirantis-jenkins commented:
Build started.

mirantis-jenkins commented:
Build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/567/

mirantis-jenkins commented:
Build finished.

mirantis-jenkins commented:
:-1: Test FAILed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_pullrequest_systest/129/
